docs/design/vm_selective_velero_restore_mechanism.md

```diff
@@ -0,0 +1,205 @@
+
+In VM file restore workflow required by the VirtualMachineFileRestore (VMFR), it is necessary to restore only the data volumes (PVs/PVCs) used by a Virtual Machine, without restoring the VM object itself nor any additional objects.
+
+Challenges:
+ - VM objects are often GitOps-managed and must not be restored.
+ - Volumes can be shared across multiple VMs in a single Velero backup.
+ - Restore method (CSI, DataMover, native) is embedded in backup metadata and all mechanisms must be supported.
+ - Velero cannot filter by resource name. Only label selectors and resource types are supported.
+ - Kubernetes labels have limits:
+   - Max 63 characters per key/value.
+   - Max 64 labels per object.
+   - Must follow DNS-compliant syntax.
+ - Label limits must be considered. Too many VMs can exceed label count per object.
+ 
+## 3. Goals
+
+ - Only data volumes must be restored. VM and other resources must be excluded.
+ - Restored objects must be traceable to their original VM.
+ - Restore must work safely in a new temporary namespace.
+ - Restore must support all data mover types: CSI, DataMover, native.
+ - Restore must be scoped to a single target VirtualMachine.
+ - Shared volumes must be restored once.
+ - Label usage must be efficient and within Kubernetes limits.
+ - Mechanism must allow VMFR Controller to generate valid Velero Restore objects with correct filters.
+
+## 4. Proposed Solution
+
+The solution relies on enhancements to the KubeVirt Velero Plugin, which will support both backup and restore operations. These enhancements are transparent to the VMFR Controller, which will continue to use standard Velero `Backup` and `Restore` objects with the appropriate configuration.
+
+### Responsibilities
+
+**KubeVirt Velero Plugin (enhanced):**
+- Extends Velero to support selective backup and restore of KubeVirt data-only resources.
+- Handles the data management logic transparently.
+
+**VMFR Controller:**
+- Triggers restores by creating Velero `Restore` objects.
+- Provides the necessary configuration expected by the plugin to selectively restore objects associated with a given VM.
+
+### 4.1. Backup Phase: Kubevirt Velero Plugin Enhancements
+
+The existing KubeVirt Velero plugin will be enhanced to embed crucial metadata into backup objects. This metadata exists solely within the Velero backup and serves as the primary mechanism for scoping restore operations. It is never persisted to live cluster resources, and during a restore, the plugin uses it to guide the process while ensuring that the restored objects match their original state, without retaining any temporary backup-only data.
+
+### 4.1.1. Metadata Strategy (Labels and Annotations)
+
+The KubeVirt Velero plugin will extend backup and restore actions for PVs and PVCs to manage additional labels and annotations used for selective restore. 
+
+**Backup Action:**
+- Add labels to PVs and PVCs (e.g., `kubevirt.io/pvc-name`, `kubevirt.io/pv-name`) to record object identity.
```

Comment:
Note that if the PVC was created by the kubevirt, it get's an explicit label on which VM own's it, same as virt-launcher pod example:

```yaml
kubevirt.io/created-by=2624f6e0-2257-4b35-9894-7f6382b44e13
```

However, this does not apply to shared PVCs, which we need to label explicitly during backup. While we could attempt to label only the PVCs that don’t already have the KubeVirt label, it’s safer and more consistent to apply the same backup label to all relevant PVCs, avoiding potential errors from special-case logic.